packages/web/app/src/components/organization/settings/oidc-integration-section.tsx

```diff
@@ -702,6 +741,118 @@ function OIDCDefaultRoleSelector(props: {
+    mode
+    projects {
+      project {
+        id
+        slug
+      }
+      targets {
+        mode
+        targets {
+          target {
+            id
+            slug
+          }
+          services {
+            mode
+            services
+          }
+          appDeployments {
+            mode
+            appDeployments
+          }
+        }
+      }
+    }
+  }
+`);
+
+function OIDCDefaultResourceSelector(props: {
+  disabled?: boolean;
+  oidcIntegrationId: string;
+  organization: DocumentType<typeof OIDCIntegrationSection_OrganizationFragment>;
+  resourceAssignment: FragmentType<typeof OIDCDefaultResourceSelector_ResourceAssignmentFragment>;
+}) {
+  const resourceAssignment = useFragment(
+    OIDCDefaultResourceSelector_ResourceAssignmentFragment,
+    props.resourceAssignment,
+  );
+  const [_, mutate] = useMutation(OIDCDefaultResourceSelector_UpdateMutation);
+
+  const [selection, setSelection] = useState<ResourceSelection>(() => ({
+    mode: resourceAssignment.mode ?? GraphQLSchema.ResourceAssignmentModeType.All,
+    projects: (resourceAssignment.projects ?? []).map(record => ({
+      projectId: record.project.id,
+      projectSlug: record.project.slug,
+      targets: {
+        mode: record.targets.mode,
+        targets: (record.targets.targets ?? []).map(target => ({
+          targetId: target.target.id,
+          targetSlug: target.target.slug,
+          services: {
+            mode: target.services.mode,
+            services: target.services.services?.map(
+              (service): GraphQLSchema.ServiceResourceAssignmentInput => ({
+                serviceName: service,
+              }),
+            ),
+          },
+          appDeployments: {
+            mode: target.appDeployments.mode,
+            appDeployments: target.appDeployments.appDeployments?.map(
+              (appDeploymentName): GraphQLSchema.AppDeploymentResourceAssignmentInput => ({
+                appDeployment: appDeploymentName,
+              }),
+            ),
+          },
+        })),
+      },
+    })),
+  }));
+
+  const [isMutating, setIsMutating] = useState<boolean>(false);
+
+  const debouncedMutate = useDebouncedCallback(
+    async (args: Parameters<typeof mutate>[0]) => {
+      setIsMutating(true);
+      await mutate(args).finally(() => {
+        setIsMutating(false);
+      });
+    },
+    1500,
+    { leading: false },
+  );
+
+  const _setSelection = useCallback(
+    async (resources: ResourceSelection) => {
+      setSelection(resources);
+      await debouncedMutate({
+        input: {
+          oidcIntegrationId: props.oidcIntegrationId,
+          resources: resourceSlectionToGraphQLSchemaResourceAssignmentInput(resources),
+        },
+      });
+    },
+    [debouncedMutate, setSelection, props.oidcIntegrationId],
+  );
+
+  return (
+    <div className="relative">
+      {(debouncedMutate.isPending() || isMutating) && (
```

Comment:
`debouncedMutate.isPending()` is true if the debounce is waiting.
This way the user gets feedback more quickly that something is happening, even though it doesn't represent the exact response times.

Reply:
Minor note here, even though I don't think its that critical, is that we are mixing "auto-save" and "click-to-save" elements here. Maybe sticking to click to save for everything might be less confusing - but not 100% sure

Reply:
I completely agree, but I think then we'd need to do the same for the existing role selector inputs, and also potentially modify the API to avoid making 3 mutations all together. I think this is better done in a future revamp of this flow. Currently, I think it's way too much information in a single modal window.